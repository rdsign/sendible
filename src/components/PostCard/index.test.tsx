import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { PostCard } from './index';
import { Template } from '../../../spec/template';

describe('PostCard', () => {
    const onDelete = jest.fn();
    const onEdit = jest.fn();

    afterEach(() => {
        jest.clearAllMocks();
    });

    const component = (
        <Template>
            <PostCard
                date="25/12/2021"
                key="c020fcff-11e3-480d-9a6d-6ebb5772dd0d"
                id="c020fcff-11e3-480d-9a6d-6ebb5772dd0d"
                imageUrl="https://images.unsplash.com/photo-1499781350541-7783f6c6a0c8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1845&q=80"
                message="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam blandit lacus neque, id commodo erat porta id. Praesent vitae dolor metus. Phasellus sed dolor et purus dapibus pulvinar vel ac mauris. Sed at euismod massa, sit amet ultrices lorem. Nulla commodo a massa et luctus."
                time="16:30"
                del={onDelete}
                edit={onEdit}
            />
        </Template>
    );

    test('should match snapshot', () => {
        const snapshot = create(component);
        expect(snapshot).toMatchSnapshot();
    });

    test('should call onEdit upon click edit button and onDelete upon click delete button', () => {
        render(component);
        const button = screen.getAllByRole('button');

        userEvent.click(button[0]);

        expect(onEdit).toHaveBeenCalledTimes(1);

        userEvent.click(button[1]);

        expect(onDelete).toHaveBeenCalledTimes(1);
    });
});
