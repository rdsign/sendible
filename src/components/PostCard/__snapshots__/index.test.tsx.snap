// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard should match snapshot 1`] = `
.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border-radius: 5rem;
  color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.6rem;
  padding: 1.4rem 2.6rem;
  -webkit-transition: all 1s cubic-bezier(0.23,1,0.32,1);
  transition: all 1s cubic-bezier(0.23,1,0.32,1);
}

.c7 svg {
  fill: #ffffff;
  height: auto;
  margin-right: 1rem;
  opacity: 0.5;
  width: 1.8rem;
}

.c7:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c0 {
  background: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 1.6rem 4.8rem rgba(0,0,0,0.07);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 2rem 0;
  height: 45rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  overflow: hidden;
  padding: 4rem;
  position: relative;
  width: 30%;
}

.c0:hover .actions {
  opacity: 1;
  top: 0;
}

.c0:hover .bg {
  -webkit-animation: spawn 2s ease-in-out 1;
  animation: spawn 2s ease-in-out 1;
  opacity: 1;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 9;
  -ms-flex: 9;
  flex: 9;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 p {
  display: -webkit-box;
  font-size: 1.6rem;
  line-height: 1.5em;
  overflow: hidden;
  -webkit-line-clamp: 7;
  -webkit-box-orient: vertical;
}

.c2 {
  border-radius: 0.8rem;
  height: 12rem;
  overflow: hidden;
  width: 100%;
}

.c2 img {
  height: 100%;
  object-fit: cover;
  width: 100%;
}

.c3 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 3rem;
}

.c4 svg {
  fill: #4d36ff;
  height: auto;
  margin-right: 1rem;
  opacity: 0.5;
  width: 1.8rem;
}

.c4 p {
  color: #4d36ff;
  font-size: 1.6rem;
  font-weight: 600;
}

.c5 {
  height: 100%;
  left: 0;
  position: absolute;
  top: 101%;
  -webkit-transition: all 1s cubic-bezier(0.23,1,0.32,1);
  transition: all 1s cubic-bezier(0.23,1,0.32,1);
  width: 100%;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #ff6153;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 9.5rem;
  left: 0;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: absolute;
  width: 100%;
  z-index: 1;
}

.c6 span {
  background: #ffffff;
  height: 100%;
  opacity: 0.3;
  width: 1px;
}

.c6 button {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.8rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 button svg {
  width: 2.4rem;
}

.c8 {
  background: linear-gradient(180deg,rgba(255,97,83,0.3) 0%,#FF6153 100%);
  top: 0;
  height: 100%;
  left: 0;
  opacity: 0;
  position: absolute;
  -webkit-transition: all 1s cubic-bezier(0.23,1,0.32,1);
  transition: all 1s cubic-bezier(0.23,1,0.32,1);
  width: 100%;
}

@media all and (min-width:260px) and (max-width:767px) {
  .c0 {
    box-shadow: 0 0.8rem 2.4rem rgba(0,0,0,0.07);
    width: 100%;
  }
}

@media all and (min-width:768px) and (max-width:1024px) {
  .c0 {
    box-shadow: 0 0.8rem 2.4rem rgba(0,0,0,0.07);
    width: 47.5%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam blandit lacus neque, id commodo erat porta id. Praesent vitae dolor metus. Phasellus sed dolor et purus dapibus pulvinar vel ac mauris. Sed at euismod massa, sit amet ultrices lorem. Nulla commodo a massa et luctus.
    </p>
    <div
      className="c2"
    >
      <img
        alt="post"
        src="https://images.unsplash.com/photo-1499781350541-7783f6c6a0c8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1845&q=80"
      />
    </div>
  </div>
  <div
    className="c3"
  >
    <div
      className="c4"
    >
      <svg
        data-testid="svg"
        fill="inherit"
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M3.26531 20C3.26531 10.7577 10.7577 3.26531 20 3.26531C29.2423 3.26531 36.7347 10.7577 36.7347 20C36.7347 29.2423 29.2423 36.7347 20 36.7347C10.7577 36.7347 3.26531 29.2423 3.26531 20ZM20 0C8.95431 0 0 8.95431 0 20C0 31.0457 8.95431 40 20 40C31.0457 40 40 31.0457 40 20C40 8.95431 31.0457 0 20 0ZM21.6327 9.79592C21.6327 8.89423 20.9017 8.16327 20 8.16327C19.0983 8.16327 18.3673 8.89423 18.3673 9.79592V20C18.3673 20.433 18.5394 20.8483 18.8455 21.1545L24.968 27.2769C25.6056 27.9145 26.6393 27.9145 27.2769 27.2769C27.9145 26.6393 27.9145 25.6056 27.2769 24.968L21.6327 19.3237V9.79592Z"
          fillRule="evenodd"
        />
      </svg>
      <p>
        25/12/2021
      </p>
    </div>
    <div
      className="c4"
    >
      <svg
        data-testid="svg"
        fill="inherit"
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M3.26531 20C3.26531 10.7577 10.7577 3.26531 20 3.26531C29.2423 3.26531 36.7347 10.7577 36.7347 20C36.7347 29.2423 29.2423 36.7347 20 36.7347C10.7577 36.7347 3.26531 29.2423 3.26531 20ZM20 0C8.95431 0 0 8.95431 0 20C0 31.0457 8.95431 40 20 40C31.0457 40 40 31.0457 40 20C40 8.95431 31.0457 0 20 0ZM21.6327 9.79592C21.6327 8.89423 20.9017 8.16327 20 8.16327C19.0983 8.16327 18.3673 8.89423 18.3673 9.79592V20C18.3673 20.433 18.5394 20.8483 18.8455 21.1545L24.968 27.2769C25.6056 27.9145 26.6393 27.9145 27.2769 27.2769C27.9145 26.6393 27.9145 25.6056 27.2769 24.968L21.6327 19.3237V9.79592Z"
          fillRule="evenodd"
        />
      </svg>
      <p>
        16:30
      </p>
    </div>
  </div>
  <div
    className="c5 actions"
  >
    <div
      className="c6"
    >
      <button
        className="c7"
        color="white"
        label="Edit"
        onClick={[Function]}
        type="button"
      >
        <svg
          data-testid="svg"
          fill="inherit"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M3.26531 20C3.26531 10.7577 10.7577 3.26531 20 3.26531C29.2423 3.26531 36.7347 10.7577 36.7347 20C36.7347 29.2423 29.2423 36.7347 20 36.7347C10.7577 36.7347 3.26531 29.2423 3.26531 20ZM20 0C8.95431 0 0 8.95431 0 20C0 31.0457 8.95431 40 20 40C31.0457 40 40 31.0457 40 20C40 8.95431 31.0457 0 20 0ZM21.6327 9.79592C21.6327 8.89423 20.9017 8.16327 20 8.16327C19.0983 8.16327 18.3673 8.89423 18.3673 9.79592V20C18.3673 20.433 18.5394 20.8483 18.8455 21.1545L24.968 27.2769C25.6056 27.9145 26.6393 27.9145 27.2769 27.2769C27.9145 26.6393 27.9145 25.6056 27.2769 24.968L21.6327 19.3237V9.79592Z"
            fillRule="evenodd"
          />
        </svg>
        Edit
      </button>
      <span />
      <button
        className="c7"
        color="white"
        label="Delete"
        onClick={[Function]}
        type="button"
      >
        <svg
          data-testid="svg"
          fill="inherit"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M3.26531 20C3.26531 10.7577 10.7577 3.26531 20 3.26531C29.2423 3.26531 36.7347 10.7577 36.7347 20C36.7347 29.2423 29.2423 36.7347 20 36.7347C10.7577 36.7347 3.26531 29.2423 3.26531 20ZM20 0C8.95431 0 0 8.95431 0 20C0 31.0457 8.95431 40 20 40C31.0457 40 40 31.0457 40 20C40 8.95431 31.0457 0 20 0ZM21.6327 9.79592C21.6327 8.89423 20.9017 8.16327 20 8.16327C19.0983 8.16327 18.3673 8.89423 18.3673 9.79592V20C18.3673 20.433 18.5394 20.8483 18.8455 21.1545L24.968 27.2769C25.6056 27.9145 26.6393 27.9145 27.2769 27.2769C27.9145 26.6393 27.9145 25.6056 27.2769 24.968L21.6327 19.3237V9.79592Z"
            fillRule="evenodd"
          />
        </svg>
        Delete
      </button>
    </div>
  </div>
  <div
    className="c8 bg"
  />
</div>
`;
